controllers/storagecluster/external_resources.go

```diff
@@ -354,20 +354,21 @@ func (r *StorageClusterReconciler) createExternalStorageClusterResources(instanc
 			} else if d.Name == cephRgwStorageClassName {
 				rgwEndpoint := d.Data[externalCephRgwEndpointKey]
 				if err := checkEndpointReachable(rgwEndpoint, 5*time.Second); err != nil {
-					r.Log.Error(err, "RGW endpoint is not reachable.", "RGWEndpoint", rgwEndpoint)
-					return err
+					r.Log.Error(err, "RGW endpoint is not reachable. Will not create objectstore and RGW Storage Class", "RGWEndpoint", rgwEndpoint)
 				}
-				extCephObjectStores, err = r.newExternalCephObjectStoreInstances(instance, rgwEndpoint)
-				if err != nil {
-					return err
+				if err == nil {
+					extCephObjectStores, err = r.newExternalCephObjectStoreInstances(instance, rgwEndpoint)
+					if err != nil {
+						return err
+					}
+					// rgw-endpoint is no longer needed in the 'd.Data' dictionary,
+					// and can be deleted
+					// created an issue in rook to add `CephObjectStore` type directly in the JSON output
+					// https://github.com/rook/rook/issues/6165
```

Comment:
```suggestion
					// rgw-endpoint is no longer needed in the 'd.Data' dictionary,
					// and can be deleted
					// created an issue in rook to add `CephObjectStore` type directly in the JSON output
					// https://github.com/rook/rook/issues/6165
```
@parth-gr could look at this comment if it still valid?

Reply:
Yes look kinda an improvement for the code 

It would need changes in both rook and oc-op